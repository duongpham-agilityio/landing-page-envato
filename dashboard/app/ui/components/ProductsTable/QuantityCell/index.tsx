// Libs
import { memo } from 'react';
import { Text, Td } from '@chakra-ui/react';

interface Props {
  quantity: number;
}

const QuantityCell = ({ quantity }: Props) => (
  <Td
    py={5}
    pr={5}
    pl={0}
    fontSize="md"
    color="text.primary"
    fontWeight="semibold"
    textAlign="left"
    w={{ base: 100, md: 20 }}
  >
    <Text
      fontSize="md"
      fontWeight="semibold"
      whiteSpace="break-spaces"
      noOfLines={1}
      w={{ base: 100, '3xl': 150, '5xl': 200 }}
      flex={1}
    >
      {quantity}
    </Text>
  </Td>
);

const QuantityCellMemorized = memo(QuantityCell);

export default QuantityCellMemorized;
