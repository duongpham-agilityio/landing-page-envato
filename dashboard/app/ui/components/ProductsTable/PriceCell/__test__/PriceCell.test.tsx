import { render } from '@testing-library/react';
import { Table } from '@chakra-ui/react';

// Components
import QuantityCell from '..';

const mockProps = {
  price: 30.5,
};

describe('QuantityCell component', () => {
  const setup = () =>
    render(<QuantityCell {...mockProps} />, {
      wrapper: Table,
    });

  it('should match with snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
