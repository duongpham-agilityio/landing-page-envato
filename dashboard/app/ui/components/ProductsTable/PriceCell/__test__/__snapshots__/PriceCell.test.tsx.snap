// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityCell component should match with snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-oe3t8o"
    >
      <p
        class="chakra-text css-13lljhc"
      >
        30.5
      </p>
    </td>
  </table>
</div>
`;
